Index: lisp/c/eus.c
===================================================================
--- lisp/c/eus.c	(revision 421)
+++ lisp/c/eus.c	(working copy)
@@ -1083,16 +1083,26 @@
 register context *ctx;
 { pointer argv, p, in;
   int i,j;
-  char *eusdir, *eusrt;
+  char *eusdir, *archdir, *eusrt;
   char fname[1024];
   extern pointer SRCLOAD();
 
   /* reset stack pointer and frame pointers*/
   j=(int)eussetjmp(topjbuf);  
-  eusdir=(char *)getenv("EUSDIR");
-  if (eusdir==NULL) { 
-    fprintf(stderr, "EUSDIR is not setenved, assuming /usr/local/eus\n");
-    eusdir= "/usr/local/eus/"; }
+  eusdir=(char *)getenv("EUSDIR");
+  archdir= "DEFAULT_ARCHDIR";
+  setenv("ARCHDIR",archdir,1);
+  char exepath[1024], *binpath; int ret;
+  ret = readlink("/proc/self/exe", exepath, 1024);
+  if ( ret > 0 ) {
+    exepath[ret] = 0;
+    binpath=dirname(strdup(exepath));
+    eusdir= dirname(dirname(strdup(binpath))); // EUSDIR/ARCHDIR/bin/<executable>
+    setenv("EUSDIR",eusdir,1);
+  } else if (eusdir==NULL) {
+    fprintf(stderr, "EUSDIR is not setenved, assuming /usr/local/eus\n");
+    eusdir= "/usr/local/eus/";
+  }
   sprintf(fname,"%s/lib/eus.init.l", eusdir);
 
 #if !Solaris2 || GCC
