Index: lisp/c/eus.h
===================================================================
--- lisp/c/eus.h	(revision 535)
+++ lisp/c/eus.h	(working copy)
@@ -137,10 +137,10 @@
 	 fib(29) nealy equals to 2.4M,
 	 fib(30) to 4M words*/
 #define MAXTHRBUDDY	6	/*small free cells cached thread-locally*/
-#define MAXSTACK	65536	/*can be expanded by sys:newstack*/
+#define MAXSTACK	8388608	/* patched: default stack size 16bit -> 23bit, can be expanded by sys:newstack*/
 #define SYMBOLHASH	60	/*initial obvector size in package*/
 #define MAXCLASS	4096	/* by M.Inaba from 64 */
-#define KEYWORDPARAMETERLIMIT 32	/*determined by bits in a long word*/
+#define KEYWORDPARAMETERLIMIT 128	/* patched: default keyword size 32->128 */
 #define ARRAYRANKLIMIT	7	/*minimal requirement for CommonLisp*/
 #define MAXTHREAD	64	/*maximum number of threads*/
 #define MAX_SPECIALS	512	/*maximum number of special variables*/
